bindings/java/src/org/sleuthkit/datamodel/CaseDbAccessManager.java

```diff
@@ -675,5 +723,226 @@ private void validateSQL(String sql) throws TskCoreException {
 		 * TODO (JIRA-5950): Need SQL injection defense in CaseDbAccessManager 
 		 */
 	}
+	
+	/**
+	 * A wrapper around a PreparedStatement to execute queries against the
+	 * database.
+	 */
+	@Beta
+	public static class CasePreparedStatement implements AutoCloseable {
+
+		private final CaseDbConnection connection;
+		private final PreparedStatement preparedStatement;
+
+		/**
+		 * Main constructor.
+		 *
+		 * @param connection        The db connection.
+		 * @param preparedStatement The delegate prepared statement.
+		 */
+		CasePreparedStatement(CaseDbConnection connection, PreparedStatement preparedStatement) {
+			this.connection = connection;
+			this.preparedStatement = preparedStatement;
+		}
+
+		/**
+		 * Returns the delegate prepared statement.
+		 *
+		 * @return The delegate prepared statement.
+		 */
+		PreparedStatement getStatement() {
+			return preparedStatement;
+		}
+
+		/**
+		 * Returns the connection to use the prepared statement with.
+		 *
+		 * @return The connection to use the prepared statement with.
+		 */
+		CaseDbConnection getConnection() {
+			return connection;
+		}
+
+		/**
+		 * Resets the parameters in the prepared statement.
+		 *
+		 * @throws TskCoreException
+		 */
+		public void reset() throws TskCoreException {
+			try {
+				preparedStatement.clearParameters();
+			} catch (SQLException ex) {
+				throw new TskCoreException("An error occurred while clearing parameters.", ex);
```

Comment:
Some of the exception messages use periods, some do not. It's a trivial point, but consistency is good. I recommend omitting the periods, since it encourages short error messages and I think the periods look funny in the logs, but that's strictly personal preference, of course.